aws-java-workshop/src/main/java/com/emc/vipr/s3/sample/_08_CreateLargeObject.java

```diff
@@ -32,69 +34,221 @@ public static void main(String[] args) throws Exception {
-    	                                                    AWSS3Factory.S3_BUCKET, key);
-    	InitiateMultipartUploadResult initResponse = 
-    	                              s3.initiateMultipartUpload(initRequest);
-
-    	// get the file and file length
-    	File file = new File(filePath);
-    	long contentLength = file.length();
-    	
-    	System.out.println( String.format("starting mulit-part upload for object [%s/%s] with file path [%s] and size [%d] in [%d] size chunks ",
-    			AWSS3Factory.S3_BUCKET, key, filePath, contentLength, partSize / 1024 / 1024));
-
-    	try {
-    	    // Step 2: Upload parts.
-    	    long filePosition = 0;
-    	    for (int i = 1; filePosition < contentLength; i++) {
-    	        // get the size of the chunk.  Note - the last part can be less than the chunk size
-    	    	partSize = Math.min(partSize, (contentLength - filePosition));
-    	    	
-    	    	System.out.println( String.format("Sending chunk [%d] starting at position [%d]", i, filePosition));
-    	    	
-    	        // Create request to upload a part.
-    	        UploadPartRequest uploadRequest = new UploadPartRequest()
-    	            .withBucketName(AWSS3Factory.S3_BUCKET).withKey(key)
-    	            .withUploadId(initResponse.getUploadId()).withPartNumber(i)
-    	            .withFileOffset(filePosition)
-    	            .withFile(file)
-    	            .withPartSize(partSize);
-
-    	        // Upload part and add response to our list.
-    	        PartETag eTagPart = s3.uploadPart(uploadRequest).getPartETag();
-    	        partETags.add(eTagPart);
-
-    	        // set file position to the next part in the file
-    	        filePosition += partSize;
-    	    }
-
-    	    // Step 3: Complete.
-    	    System.out.println("Waiting for completion of multi-part upload");
-    	    CompleteMultipartUploadRequest compRequest = new 
-    	                CompleteMultipartUploadRequest(AWSS3Factory.S3_BUCKET,
-    	                                               key, 
-    	                                               initResponse.getUploadId(), 
-    	                                               partETags);
-
-    	    s3.completeMultipartUpload(compRequest);
-    	} catch (Exception e) {
-    	    s3.abortMultipartUpload(new AbortMultipartUploadRequest(
-    	    		AWSS3Factory.S3_BUCKET, key, initResponse.getUploadId()));
-    	}
-
-        // print bucket key/value and content for validation
-    	System.out.println( String.format("completed mulit-part upload for object [%s/%s] with file path: [%s]",
-    			AWSS3Factory.S3_BUCKET, key, filePath));
+
+	_08_CreateLargeObject clo = new _08_CreateLargeObject();
+	clo.mpuSingleThreaded(key, filePath);
+	//clo.testMultiThreadMultipartUploadListPartsPagination(key, filePath);
     }
+
+    public void mpuSingleThreaded(String key, String filePath) {
+		AmazonS3 s3 = AWSS3Factory.getS3Client();
```

Comment:
I'm not talking about the reference in the main method.  The reference in the main method can be removed entirely.  Right?  Why is there an AmazonS3 client being created in the main method?  Can you not use the variable _client_ for every call to S3?  Why does each method need to create it's on AmazonS3 client variable?